### PR TITLE
pythonPackages.haas-nabucasa: init at 0.29

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Things done

- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/python-modules/snitun/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    rev = version;
+    sha256 = "1nscfwycclfbll709w1q46w6rl0r5c3b85rsc7zwc3ixd1k8aajp";
+  };
+
+  propagatedBuildInputs = [ attrs cryptography async-timeout ];
+
+  checkInputs = [ pytest pytest-aiohttp ];
+
+  checkPhase = ''
+    ${python.interpreter} -m pytest tests/
```
Comment (jonringer) [inline pkgs/development/python-modules/snitun/default.nix]: the pytest package also exports an executable that you can call as well
```suggestion
    pytest tests/
```

File: pkgs/development/python-modules/snitun/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.nabucasa.com/";
```
Comment (jonringer) [inline pkgs/development/python-modules/snitun/default.nix]: personally i would prefer the repo:
```suggestion
    homepage = "https://github.com/nabucasa/snitun";
```

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
has tests

```
[4 built, 2 copied (0.0 MiB), 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/75876
2 package were built:
python37Packages.snitun python38Packages.snitun
```

just some small suggestions

**Comment by jonringer** [comment]
@GrahamcOfBorg build python37Packages.snitun python38Packages.snitun

**Comment by Scriptkiddi** [comment]
@jonringer I added another package to this pr since it depends on the original one, or should I open a new pull request for the next one?

**Review by jonringer — COMMENTED** [review]
your commit history should be:
```
pythonPackages.snitun: init at 0.20
pythonPackages.haas-nabucasa: init at 0.29
```
I would recommend using `git rebase -i`

**Comment by jonringer** [comment]
it's fine, just need to update your PR to describe what you're doing

**Comment by jonringer** [comment]
you have snitun changes in the pythonPackages.haas-nabucasa commit do the following to fix:

```
git reset HEAD~2 # un-commit last 2 commits, and put in unstaged
git add pkgs/development/python-modules/snitun/default.nix
git commit -m "pythonPackages.snitun: init at 0.20"
git add pkgs/development/python-modules/snitun/default.nix
git commit -m "pythonPackages.haas-nabucasa: init at 0.29"
git push <fork> <branch> --force
```

**Comment by flokli** [comment]
@Scriptkiddi can this be moved inside the `pkgs/servers/home-assistant` folder?

As elaborated in https://github.com/NabuCasa/hass-nabucasa/pull/119#issuecomment-572033883, upstream "doesn't care about other ways to run Home Assistant" (except docker and virtualenv install), so we should probably not have it in the curated python package set, but move it to https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/home-assistant/default.nix, (with some overrides, so this module doesn't block `acme` version bumps).

**Comment by Scriptkiddi** [comment]
I will do that, I will also see if we can not remove such a module upstream from the hass cloud module so its not required, lets see how good my chances are seeing that those guys from nabucasa started hass ....

**Pull request merged**